dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -604,5 +622,112 @@ public String getUrl(Contentlet contentlet) {
+    public void Test_Serialize_Contentlet_Then_Recover_Then_Transform(final SerializationTestCase serializationTestCase)
+            throws IOException, ClassNotFoundException {
+
+        final File file = new File(directory,String.format("%s.serialized",System.currentTimeMillis()));
+        file.createNewFile();
+
+        final Contentlet original = serializationTestCase.contentlet;
+        serializeContentlet(serializationTestCase.contentlet, file);
+        final Contentlet copy = readSerializedContentlet(file);
+        validateMaps(original, copy);
+        final AssertionStrategy assertionStrategy = serializationTestCase.assertionStrategy;
+        if(null != assertionStrategy) {
+            assertionStrategy.apply(original, copy);
+        }
+    }
+
+    private static class SerializationTestCase {
+
+        final Contentlet contentlet;
+        final AssertionStrategy assertionStrategy;
+
+        SerializationTestCase(final Contentlet contentlet,
+                final AssertionStrategy assertionStrategy) {
+            this.contentlet = contentlet;
+            this.assertionStrategy = assertionStrategy;
+        }
+    }
+
+    @DataProvider
+    public static Object[] listSerializeTestCases() throws Exception {
+        final FileAssetAPI assetAPI = APILocator.getFileAssetAPI();
+        HTMLPageAssetAPI pageAssetAPI = APILocator.getHTMLPageAssetAPI();
+        return new Object[]{
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getFileAssetSVGContent(true, langId);
+                    return assetAPI.fromContentlet(contentlet);
+                }).get(), fileAssetValidation),
+                new SerializationTestCase(((Supplier<Contentlet>) () -> {
+                    final Contentlet contentlet = TestDataUtils.getPageContent(true, langId);
+                    return pageAssetAPI.fromContentlet(contentlet);
+                }).get(),
+                        (original, copy) -> {
+
+                        })
+
+        };
+    }
+
+    private void validateMaps(final Contentlet original, final Contentlet copy){
+        final Map<String, Object> preSerializationMap = new DotTransformerBuilder().defaultOptions().content(original).build().toMaps().get(0);
+        final Map<String, Object> postSerializationMap = new DotTransformerBuilder().defaultOptions().content(copy).build().toMaps().get(0);
+        Assert.assertEquals(preSerializationMap, postSerializationMap);
+    }
+
+    private static final AssertionStrategy fileAssetValidation = (original, copy) -> {
+        final FileAsset fileAsset1 = (FileAsset) original;
+        final int width = fileAsset1.getWidth();
+        final int height = fileAsset1.getHeight();
+
+        final FileAsset fileAsset2 = (FileAsset) copy;
+
+        final int width2 = fileAsset2.getWidth();
+        final int height2 = fileAsset2.getHeight();
+        Assert.assertEquals(width, width2);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5883765)